model/riscv_mem.sail

```diff
@@ -218,12 +229,12 @@ function checked_mem_write forall 'n, 0 < 'n <= max_mem_access . (
     Some(e) => Err(e),
     None() => {
```

Comment:
Wouldn't it be better to handle the endianness conversion here? Then you don't need to handle it separately for `mmio_write` and `phys_mem_write`.